lua/rip-substitute/utils.lua

```diff
@@ -1,6 +1,26 @@
 local M = {}
 --------------------------------------------------------------------------------
 
+---@generic T
+---@param list Array<`T`>
+---@param cb function(value: `T`,i:integer): `T`
+M.map = function(list, cb)
```

Comment:
1. For consistency with the rest of the repo, it should be `function M.map`
2. Is there really a need for a map-util, when nvim already offers us `vim.tbl_map` and `Iter:map()`? If so, the explanation why a custom map implementation is needed would be something that should be noted in a comment

Reply:
using vim.tbl_map now